common/lib/dependabot/config/ignore_condition.rb

```diff
@@ -48,53 +48,63 @@ def versions_by_type(dependency)
 
       def ignore_patch(version)
         parts = version.split(".")
-        return [] unless parts.size > 2
-
-        lower_parts = parts.first(2) + ["a"]
-        upper_parts = parts.first(2)
-        upper_parts[1] = upper_parts[1].to_i + 1
+        version_parts = parts.fill(0, parts.length...4)
+        lower_parts = if numeric_version?(version)
```

Comment:
Good question! Digging into this is making me feel like this is more complicated than it needs to be. I'm going to take another pass and look at simplifying. I think we don't need to be too concerned about the behavior when non semver versions are provided.

Reply:
Ok, I've simplified this a bit. Instead of trying to deal with non ruby gem compatible versions we bail early with no results. We'd still do weird things with a version like `1.banana` but I don't think that should end up breaking anything.

I also found the attempt to use `.a` in the ignore-patch version ranges was causing issues with pre-release semver versions like `1.2.3-alpha.1`. I've reverted back to the original attempt which used `>` instead of `>=` & `.a` in the ignore range.